### PR TITLE
Update Repo Structure for AppModelConfig and AzureCore

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Moved ApplicationModelConfiguration and Core to sdk folder.
Updated sdk structure

### CODE REVIEW

**Comment by chidozieononiwu** [comment]
@Azure/azure-sdk-eng

**Comment by weshaggard** [comment]
FYI @pakrym @AlexGhiondea @maririos 

@mariros do you know what the config service is called? Is it just azconfig or is there a different name? We want to know what to name the service folder that will contain it.

File: sdk/appconfiguration/Azure.ApplicationModel.Configuration/Performance/Azure.ApplicationModel.Configuration.Performance.csproj [hunk 1]
```diff
@@ -11,10 +11,10 @@
   </ItemGroup>
```
Comment (weshaggard) [inline sdk/appconfiguration/Azure.ApplicationModel.Configuration/Performance/Azure.ApplicationModel.Configuration.Performance.csproj]: This project should be moved under the tests folder.

**Comment by maririos** [comment]
> @mariros do you know what the config service is called? Is it just azconfig or is there a different name? We want to know what to name the service folder that will contain it.

The name of the service is `App Configuration`
As for the namespace we are going to use for it, it was just decided in the arch board meeting that it will be `Azure.Data.AppConfiguration` =>changes to the code to reflect that namespace will come during this week

**Comment by weshaggard** [comment]
>The name of the service is App Configuration

Thanks. @chidozieononiwu then what you have as appconfiguration as the name for the service folder is correct.

File: sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <PackageReference Include="System.Memory" Version="4.5.2" />
   </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\Azure.Core\Azure.Core.csproj">
+    <ProjectReference Include="..\src\Azure.Core.csproj">
       <Project>{FEC0AE57-E241-46DB-B7C7-9181F28E1D3F}</Project>
       <Name>Azure.Core</Name>
     </ProjectReference>
```
Comment (weshaggard) [inline sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: @maririos at some point we will want to move the samples up a level into a samples directory parallel to the tests directory.

Comment (maririos) [reply sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: Ohh interesting. Is it part of the repo structure guidelines and we just missed it?

Comment (weshaggard) [reply sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: Yes see. https://github.com/Azure/azure-sdk/blob/master/docs/engineering-system/repo-structure.md we hope to make the samples individually runnable separate from our tests. We also hope to try and tie them into the sample code in our README's for the libraries.

**Review by weshaggard — APPROVED** [review]

File: sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj (hunk 2 above)
Comment (maririos) [reply sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: I see. Issue https://github.com/Azure/azure-sdk-for-net/issues/6085 created to address that.

As for the Readme, we do add code snippets from the Hello World sample, following the [Reame Guidelines](https://github.com/Azure/azure-sdk/blob/master/docs/README-TEMPLATE.md)

**Pull request merged**